sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -478,12 +479,11 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
     while (buf.length < n && partsScanned < totalParts) {
       // The number of partitions to try in this iteration. It is ok for this number to be
       // greater than totalParts because we actually cap it at totalParts in runJob.
-      var numPartsToTry = 1L
+      var numPartsToTry = Math.max(conf.initialNumPartitions, 1)
       if (partsScanned > 0) {
         // If we didn't find any rows after the previous iteration, quadruple and retry.
```

Comment:
Just realized that this comment (and the ones in the other file) have been outdated ever since we originally introduced the `limitScaleUpFactor` configuration. Maybe we can change "quadruple and retry" to "multiply by limitScaleUpFactor and retry"?